rest/counted_response_writer.go

```diff
@@ -92,3 +92,10 @@ func (w *CountedResponseWriter) isHijackable() bool {
 	_, ok := w.writer.(http.Hijacker)
 	return ok
 }
+
+func (w *CountedResponseWriter) Flush() {
```

Comment:
I think we should add some compile time assertions for all the interfaces that we implement them.

```
var _ http.Flusher = &CountedResponseWriter{}
var _ http.Flusher = &NonCountedResponseWriter{}
var _ http.Flusher = &EncodedResponseWriter{}
```

while we are there, I think we should assert on a hijacker interface, since I made that mistake too, before.

```
var _ http.Hijacker = &CountedResponseWriter{}
var _ http.Hijacker = &NonCountedResponseWriter{}
var _ http.Hijacker = &EncodedResponseWriter{}
```